### DISCUSSION TITLE
Support for MPU6500

### DISCUSSION THREAD

**Discussion by Bananamannn** [mp-se/gravitymon#73] [Ideas]
Hi. Would it be possible to add support for the MPU6500 as well as the MPU6050? They both appear to have the same operational values but have different addresses 0x34 for the 6050 and 0x38 for the 6500.

**Comment by mp-se** [comment]
The 6000 has the same features as the 6050 (3 axis gyro) but the i2c library that i use only supports the 6050. I dont know if hey have the same api commands and features. But all the gyro code is in one class so it could be done if there is an library for arduino. I dont have any of those chips so i cant trst either.

**Reply by Bananamannn** [reply]
I have used the 6500 with the 6050 library before and it appears to work just fine, the only thing i've found that doesn't work is the `.isconnected` function as the address is different. I have done the below code in the past to make my own connected test and it's worked well. I have 6500 chips so i'm quite happy to test it. 

```
bool testAccel()
{
  uint8_t res = Wire.status();
    if (res != I2C_OK)
    CONSOLELN(String(F("I2C ERROR: ")) + res);
  
  bool con = false;
  if (accelgyro.getDeviceID()==0x34)
  {
    con = true;
  }
  
  if (accelgyro.getDeviceID()==0x38)
  {
    con = true;
  }
  
  if (!con)
    CONSOLELN(F("Acc Test Connection ERROR!"));
      
  return res == I2C_OK && con == true;
}
```

**Reply by mp-se** [reply]
If its just about the i2c ID I can adjust the code to support that. Would you mind submitting a issue on the topic so I have it in the backlog ? You can reference this discussion in the issue.

**Reply by Bananamannn** [reply]
Sure thing. I'll add it now

**Comment by mp-se** [comment]
@Bananamannn  I've just published beta2 with this change. Let me know if it works, if not I can build a version with more verbose output for the gyro.
